cmd/argocd/commands/app.go

```diff
@@ -1804,13 +1826,51 @@ func NewApplicationWaitCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 }
 
 // printAppResources prints the resources of an application in a tabwriter table
-func printAppResources(w io.Writer, app *argoappv1.Application) {
+func printAppResources(w io.Writer, app *argoappv1.Application, discoveryClient discovery.DiscoveryInterface) {
 	_, _ = fmt.Fprintf(w, "GROUP\tKIND\tNAMESPACE\tNAME\tSTATUS\tHEALTH\tHOOK\tMESSAGE\n")
-	for _, res := range getResourceStates(app, nil) {
+	versionedResourceStates := getVersionedResourceStates(app, nil)
+	for _, res := range mergeDuplicateClusterLevelResources(versionedResourceStates, discoveryClient) {
 		_, _ = fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", res.Group, res.Kind, res.Namespace, res.Name, res.Status, res.Health, res.Hook, res.Message)
 	}
 }
 
+func mergeDuplicateClusterLevelResources(resourceStates []*versionedResourceState, discoveryClient discovery.DiscoveryInterface) []*versionedResourceState {
+	resourcesByKey := make(map[kube.ResourceKey]*versionedResourceState)
+	// 	GROUP        KIND  NAMESPACE                                      NAME               STATUS     HEALTH   HOOK  MESSAGE
+	// argoproj.io  Fake  argocd-e2e--test-app-sync-wrong-version-xunes  fake-crd-instance  OutOfSync  Missing        The Kubernetes API could not find version "v1alpha2" of argoproj.io/Fake for requested resource argocd-e2e--test-app-sync-wrong-version-xunes/fake-crd-instance. Version "v1alpha1" of argoproj.io/Fake is installed on the destination cluster.
```

Comment:
What is this comment used for?

Reply:
Oh, that was a debugging thing. The issue was the output generated by the command and it differed between my branch and master, so I copied the content here as guidance on what it should look like after I fixed the bug.

Will cleanup.